mesonbuild/interpreter.py

```diff
@@ -509,6 +510,24 @@ def get_command(self):
     def get_name(self):
         return self.held_object.get_name()
 
+    def get_version(self, interpreter):
+        if not self.cached_version:
+            raw_cmd = self.get_command() + ['--version']
+            cmd = [self, '--version']
+            res = interpreter.run_command_impl(interpreter.current_node, cmd, {}, True)
+            if res.returncode != 0:
+                m = 'Running {!r} failed'
+                raise InterpreterException(m.format(raw_cmd))
+            output = res.stdout.strip()
+            if not output:
+                output = res.stderr.strip()
+            match = re.search(r'([0-9\.]+)', output)
```

Comment:
Can we use `environment.search_version()` or is that not a good heuristic?

Reply:
The problem with that is you cannot document the algorithm. All you can say is it magically try to extract a version, with no guarantee it won't change and regress in some cases. What I like with my proposal is the algorithm is openly documented so there is no surprise.

Reply:
> The problem with that is you cannot document the algorithm

I'm not sure I understand what you're saying. The `search_version()` regex is quite well-documented. It even has `InternalTests` unit tests. Would be nice to add more tests for any other tools that have 'interesting' output.

Reply:
Actually, `environment.search_version()` also handles a case that your regex does not: `(Sourcery CodeBench Lite 2014.05-29) 4.8.3 20140320 (prerelease)` where 4.8.3 is the actual version. There's also a test for it.

Reply:
I would imagine the difference here is something that is internally documented vs externally documented. The latter can't be changed easily (or at all).

Reply:
@jpakkane exactly. If we use `search_version()` it usually does a better job, but we cannot document the exact algorithm so we could have regressions because we don't know all the cases users could have been using it for. If we use my current proposal, it doesn't work in some cases (I noticed `gst-launch-1.0` and `perl`) but at least user can rely on it will always work the same way.

Reply:
I personally don't have a strong opinion here, both are fine with me. tbh, the ouput format of `--version` is not even guaranteed to not change, so I guess it's a situation where we cannot do more than supporting "most cases".

Reply:
Sounds fine to me.